fix(client): render chapter-level review and exam as standalone link

huyenltnguyen opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

This PR:
- Updates the rendering logic of super block accordion to display chapter-level review and exam blocks as link, rather than an accordion
- Fixes some minor UI bugs

<details>
  <summary>Screenshots</summary>

| Before | After | 
| --- | --- |
| <img width="774" alt="Screenshot 2024-11-21 at 03 11 05" src="https://github.com/user-attachments/assets/9b90b5b9-0497-4076-bab7-1718b4adcb67"> | <img width="771" alt="Screenshot 2024-11-21 at 03 11 46" src="https://github.com/user-attachments/assets/101b5101-ba77-45de-84a8-19f718560073"> |

</details>

<!-- Feel free to add any additional description of changes below this line -->

huyenltnguyen committed:
fix: remove extra space below link block

huyenltnguyen committed:
fix: module title casing

huyenltnguyen committed:
fix: render chapter-level review and exam as standalone block

huyenltnguyen committed:
fix: remove bottom border from last li in chapter

moT01 commented on client/src/templates/Introduction/components/super-block-accordion.tsx:
```diff
@@ -108,42 +119,70 @@ export const SuperBlockAccordion = ({
       {allChapters.map(chapter => {
-        if (examChallenge && chapter.name === examChallenge.chapter) {
+        const chapterAsLinkChallenge = examChallenges.find(
+          challenge => challenge.dashedName === chapter.name
```
If we end up [adding this](github.com/freeCodeCamp/freeCodeCamp/pull/57152), these checks against the challenge dashed name won't work. Is there another way we can do this @huyenltnguyen? Like maybe adding a property in the `full-stack.json` file?

ojeytonwilliams replied:
Yeah, that makes sense @moT01, so I went with it.

I'm not 100% convinced importing directly from the curriculum is _good_, but I think it's okay for now.

ojeytonwilliams committed:
refactor: use superblock structure for review modules

ojeytonwilliams committed:
feat: adding chapterType

TypeScript was confused by chapterType alone, so I added another
moduleType to keep it happy.

ojeytonwilliams committed:
refactor: use superblock structure for exam chapters

ojeytonwilliams reviewed (approved):
I think this is a fair bit better, thanks @huyenltnguyen! I tried to make sure my changes slotted in with yours, so hopefully they make sense.

@moT01 do you want to get this in so there's less conflict with what you're working on?

huyenltnguyen committed:
Merge branch 'main' of https://github.com/freeCodeCamp/freeCodeCamp into fix/prep-exams

huyenltnguyen committed:
fix: update schema
